### DISCUSSION TITLE
lockFileMaintenance commitBodyTable configuration option

### DISCUSSION THREAD

**Discussion by mikestreety** [renovatebot/renovate#22565] [Suggest an Idea]
### Tell us more.

I would like a `commitBodyTable` option for the `lockFileMaintenance` configuration.

It would be great if the lockFileMaintenance commit body could include a table of what was updated. We have sub-dependencies that are potentially quite important to know if they are being updated and rather than trawling through the diff, it would be amazing if a summary table could be added like you can for other updates

e.g. 

<img width="879" alt="image" src="https://github.com/renovatebot/renovate/assets/354085/b0b1d179-bdc3-48a0-ab41-7f85bbcaa3b0">

**Comment by rarkins** [comment]
This table could be thousands of lines long and include additions, modifications, and deletions

**Reply by mikestreety** [reply]
I'm assuming it would be off by default, like the current one is? 

I imagined it having options like the `packageRules` - in that you could turn it on and off, depending if it was a major, minor or patch, and also depending on manager. Composer doesn't seem to have as many sub-dependencies as something like NPM.

Alternatively, is there a way I could flag if any of my sub-dependencies get bumped with a major or a minor version?

**Reply by rarkins** [reply]
No, it would not be something which could be controlled by packageRules and right now packages updated by lock file maintenance aren't parsed/understood by Renovate (it just commits the changes which npm/composer/etc make)

**Reply by mikestreety** [reply]
Ah right, so it essentially does a `composer update && git commit -am 'Update lockfiles' ?

Are there plans in the future to parse this or will it stay as the "catch-all"? (sorry for all the questions)

**Reply by HonkingGoose** [reply]
@mikestreety If you're on GitHub: you can view a nice diff of the dependency changes. Read the [GitHub Docs, Reviewing dependency changes in a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-dependency-changes-in-a-pull-request) to see how.

**Reply by mikestreety** [reply]
Thanks @HonkingGoose - that looks great, unfortunately we use Gitlab so that's not available.

Thanks for all your help - I might have to move some sub-dependencies into the main repo so I get more visibility on the updates

**Reply by HonkingGoose** [reply]
> Ah right, so it essentially does a `composer update && git commit -am 'Update lockfiles' ?

That's right!

> Are there plans in the future to parse this or will it stay as the "catch-all"? (sorry for all the questions)

That's up to the maintainers. I suppose doing more than "package manager updates lockfile(s) and Renovate commits whatever changed" is too difficult. Or it's too much effort to maintain that behavior for each package manager that Renovate supports. 😄

**Reply by viceice** [reply]
yes, too much work and probably too big for the PR description.

**Reply by HonkingGoose** [reply]
I'm rewriting our `lockfileMaintenance` docs to better explain _how_ Renovate updates lockfiles:

- https://github.com/renovatebot/renovate/pull/22617

**Reply by mikestreety** [reply]
Thanks for the answer everyone (and thanks for the doc update @HonkingGoose )

**Reply by HonkingGoose** [reply]
Cheers! 😄 ❤️
